create_layout.py

```diff
@@ -91,45 +91,111 @@
 import in_toto.models.link
 import in_toto.models.layout
 
-def create_material_rules(links, index):
+def changes_between_snapshots(before_dict, after_dict):
+  '''Given two 'snapshots' of a file structure -- 'before' and 'after' -- 
+    return a tuple specifying which files have been added, 
+    which have been removed, which have been modified, 
+    and which have remained unchanged. 
+    Both these dictionaries have file
+    names as the keys and their hashes as the values.'''
+
+  unchanged_files = set()
+  modified_files = set()
+  added_files = set()
+  removed_files = set()
+  for key in before_dict:
+    if key in after_dict:
+      if before_dict[key] == after_dict[key]:
+        # Matching the hashes to check if file was unchanged
+        unchanged_files.add(key)
+      else:
+        modified_files.add(key)
+    else:
+      removed_files.add(key)
+  for key in after_dict:
+    if key not in before_dict:
+      # Looking for new files
+      added_files.add(key)
+
+  # Returning the snapshot of the new file system
+  return (sorted(unchanged_files), sorted(modified_files), sorted(added_files),
+      sorted(removed_files))
+
+def create_material_rules(previous_link, current_link):
   """Create generic material rules (3 variants)
 
+  previous_link: the link of previous step
```

Comment:
I understand we're using the google docstring format as @lukpueh suggests in https://github.com/secure-systems-lab/code-style-guidelines/issues/20.